app/assets/javascripts/pages/agent-edit-page.js.coffee

```diff
@@ -122,5 +130,51 @@ class @AgentEditPage
       else
         @hideEventCreation()
 
+  enableDryRunButton: ->
+    $(".agent-dry-run-button").prop('disabled', false).off().on "click", @invokeDryRun
+
+  disableDryRunButton: ->
+    $(".agent-dry-run-button").prop('disabled', true)
+
+  invokeDryRun: (e) ->
+    e.preventDefault()
+    $(".agent-dry-run-button").prop('disabled', true)
+    $('body').css(cursor: 'progress')
+    $.ajax type: 'POST', url: $(".agent-dry-run-button").data('action-url'), dataType: 'json', data: $(@form).serialize()
+      .always =>
+        $("body").css(cursor: 'auto')
+      .done (json) =>
+        $("body").append """
+          <div class="modal fade" tabindex="-1" id='dynamic-modal' role="dialog" aria-labelledby="dynamic-modal-label" aria-hidden="true">
```

Comment:
There is a very similar block of code [here](https://github.com/cantino/huginn/blob/master/app/assets/javascripts/pages/agent-show-page.js.coffee#L22), should we combine them?